src/Analyser/NamedArgumentsHelper.php

```diff
@@ -0,0 +1,80 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Analyser;
+
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\CallLike;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\StaticCall;
+use PHPStan\Reflection\ParametersAcceptor;
+
+final class NamedArgumentsHelper
```

Comment:
Really nice! I'd like this class to have separate unit tests so that we can see what happens to those gaps between arguments if an optional one is skipped. To me it looks like this code still doesn't handle that.